### PR TITLE
Move history and matching failure metric to common

### PR DESCRIPTION
**Pull request by madhuravi**

### CODE REVIEW

File: common/metrics/defs.go [hunk 1]
```diff
@@ -535,14 +535,15 @@ const (
 	PersistenceErrBusyCounter
 
 	NumCommonMetrics
+	HistoryFailures
```
Comment (samarabbas) [inline common/metrics/defs.go]: NumCommonMetrics need to be the last enum.  Please declare the new ones before this.

File: common/metrics/defs.go [hunk 2]
```diff
@@ -620,13 +620,14 @@ var MetricDefs = map[ServiceIdx]map[int]metricDefinition{
 		PersistenceErrConditionFailedCounter:          {metricName: "persistence.errors.condition-failed", metricType: Counter},
 		PersistenceErrTimeoutCounter:                  {metricName: "persistence.errors.timeout", metricType: Counter},
 		PersistenceErrBusyCounter:                     {metricName: "persistence.errors.busy", metricType: Counter},
+		HistoryFailures:                               {metricName: "history.client.errors", metricType: Counter},
```
Comment (samarabbas) [inline common/metrics/defs.go]: can you change the name to be client.history.errors and client.matching.errors

Comment (samarabbas) [inline common/metrics/defs.go]: Also rename the scopes to be HistoryClientFailures and HistoryMatchingFailures.

File: common/metrics/defs.go (hunk 1 above)
Comment (madhuravi) [reply common/metrics/defs.go]: Why?

Comment (madhuravi) [reply common/metrics/defs.go]: Nm, I see the subsequent constants.

**Pull request merged**